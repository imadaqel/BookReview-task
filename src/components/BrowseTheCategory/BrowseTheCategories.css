/* .section {
    margin: 0;
    padding: 0;
}
.section-content {
    max-width: 800px;
    margin: auto;
    padding: 5rem 0;
}

.section-content > h1 {
    margin: 0;
    padding: 0;
}
.section-content > h1 > span {
    font-weight: lighter;
}
.categories-wrapper {
    background-color: white;
}
.categories-main {
    display: grid;
    grid-gap: 2rem;
    grid-template-columns: auto auto auto auto;
    margin-top: 2rem;
}

.categories-card {
    display: flex;
    flex-direction: row;
    min-width: 170px;
    padding: 0;
    margin: 0;
}
.categories-card > img {
    display: inline-block;
    width: 40px;
    height: 40px;
    padding: 0;
}
.categories-card-content {
    flex: 1;
    padding-left: 10px;
}
.categories-card-content > h4 {
    margin: 0;
}

.categories-card-content > p {
    margin: .25rem 0 0;
    hyphens: auto;
    overflow: hidden;
    color: gray;
}

@media only screen and (max-width: 768px) and (min-width: 425px) {
    .categories-main {
        display: grid;
        grid-gap: 0rem;
        grid-template-columns: auto auto;

    }
    .categories-card {
        max-width: 28rem;
    }
}

@media only screen and (max-width: 425px) and (min-width: 20px) {
    .categories-main {
        display: grid;
        grid-gap: 0rem;
        grid-template-columns: auto ;
        margin: 2rem 2rem;
        justify-content: start
    }
    .categories-card {
        max-width: 28rem;
        padding: 2rem;
    }
    
    .section-content > h1{
        font-size: 26px;
        text-align: center;
        }
}

@media only screen and (max-width: 320px) {} */