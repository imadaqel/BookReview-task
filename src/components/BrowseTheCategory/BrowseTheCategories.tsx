import React from 'react'
// import './BrowseTheCategories.css';
import {CategoriesWrapper,SectionContent,CategoriesHeader,SectionContentSpan,CategoriesMain,CategoriesCard,
    CategoriesCardImg,CategoriesCardContent,CategoriesMainTitle,CategoriesCardParg} from './BrowseTheCategoriesStyle'

export default function BrowseTheCategories(props: any) {
    const Categories: IItem[] = props.Categories;
    console.log(props.Categories)
    interface IItem {
        id: number;
        image: string;
        title: string;
        description: string;
    }
    return (
            <CategoriesWrapper>
                <SectionContent>
                    <CategoriesHeader>Browse<SectionContentSpan> Our Most Popular Categories</SectionContentSpan>
                    </CategoriesHeader>
                    <CategoriesMain>
                    {Categories && Categories.length > 0 && Categories.map((item: IItem) =>
                        <CategoriesCard key={item.id}>
                            <CategoriesCardImg
                            // src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iaXNvLTg4NTktMSI/Pg0KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDE5LjAuMCwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPg0KPHN2ZyB2ZXJzaW9uPSIxLjEiIGlkPSJDYXBhXzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4Ig0KCSB2aWV3Qm94PSIwIDAgNTEyIDUxMiIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgNTEyIDUxMjsiIHhtbDpzcGFjZT0icHJlc2VydmUiPg0KPGc+DQoJPGc+DQoJCTxwYXRoIGQ9Ik0yNTYsMzFDMTE1LjM5LDMxLDAsMTQ1LjM5LDAsMjg2djEyMGMwLDI0LjgxNCwyMC4xODYsNDUsNDUsNDVoMTVWMzAxSDQ1Yy01LjI4NCwwLTEwLjI4NSwxLjA4Mi0xNSwyLjc2M1YyODYNCgkJCUMzMCwxNjEuOTI4LDEzMS45MjgsNjEsMjU2LDYxczIyNiwxMDAuOTI4LDIyNiwyMjV2MTcuNzYzYy00LjcxNS0xLjY4MS05LjcxNi0yLjc2My0xNS0yLjc2M2gtMTV2MTUwaDE1DQoJCQljMjQuODE0LDAsNDUtMjAuMTg2LDQ1LTQ1VjI4NkM1MTIsMTQ1LjM5LDM5Ni42MSwzMSwyNTYsMzF6Ii8+DQoJPC9nPg0KPC9nPg0KPGc+DQoJPGc+DQoJCTxwYXRoIGQ9Ik0xMzUsMjcxaC0xNWMtMTYuNTY5LDAtMzAsMTMuNDMxLTMwLDMwdjE1MGMwLDE2LjU2OSwxMy40MzEsMzAsMzAsMzBoMTVjOC4yODQsMCwxNS02LjcxNiwxNS0xNVYyODYNCgkJCUMxNTAsMjc3LjcxNiwxNDMuMjg0LDI3MSwxMzUsMjcxeiIvPg0KCTwvZz4NCjwvZz4NCjxnPg0KCTxnPg0KCQk8cGF0aCBkPSJNMzkyLDI3MWgtMTVjLTguMjg0LDAtMTUsNi43MTYtMTUsMTV2MTgwYzAsOC4yODQsNi43MTYsMTUsMTUsMTVoMTVjMTYuNTY5LDAsMzAtMTMuNDMxLDMwLTMwVjMwMQ0KCQkJQzQyMiwyODQuNDMxLDQwOC41NjksMjcxLDM5MiwyNzF6Ii8+DQoJPC9nPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPGc+DQo8L2c+DQo8Zz4NCjwvZz4NCjxnPg0KPC9nPg0KPC9zdmc+DQo="
                            src={item.image} alt="can't load Categories Card Img "
                            >            
                            </CategoriesCardImg>
                                <CategoriesCardContent>
                    <CategoriesMainTitle>{item.title}</CategoriesMainTitle>
                    <CategoriesCardParg>{item.description}</CategoriesCardParg>
                                </CategoriesCardContent>
                        </CategoriesCard>
                    )}
                </CategoriesMain>
            </SectionContent>
        </CategoriesWrapper>
    )
}
