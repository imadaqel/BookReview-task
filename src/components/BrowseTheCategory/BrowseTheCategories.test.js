import React from 'react';
import renderer from 'react-test-renderer';
import BrowseTheCategories from './BrowseTheCategories';

it('renders BrowseTheCategories correctly when there are no items', () => {
  const tree = renderer.create(<BrowseTheCategories />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders BrowseTheCategories correctly when there are multiple items', () => {
  
    var categories = [
      {
        "id": 1,
        "image": "/categorayImages/category1.png",
        "title": "Aenean",
        "description": "Mauris sollicitudin fermentum libero. Phasellus viverra nulla ut metus varius laoreet."
      }]
  const tree = renderer.create(<BrowseTheCategories categories={categories} />).toJSON();
  expect(tree).toMatchSnapshot();
});