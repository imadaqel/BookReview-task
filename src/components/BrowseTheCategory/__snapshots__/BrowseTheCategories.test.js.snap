// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are multiple items 1`] = `
<section
  className="sc-bdVaJa hwOTrO"
>
  <section
    className="sc-bwzfXH hDJlEA"
  >
    <h1
      className="sc-htpNat dPDbWX"
    >
      Browse
      <span
        className="sc-bxivhb hfKSal"
      >
         Our Most Popular Categories
      </span>
    </h1>
    <section
      className="sc-ifAKCX iUhJXj"
    />
  </section>
</section>
`;

exports[`renders correctly when there are no items 1`] = `
<section
  className="sc-bdVaJa hwOTrO"
>
  <section
    className="sc-bwzfXH hDJlEA"
  >
    <h1
      className="sc-htpNat dPDbWX"
    >
      Browse
      <span
        className="sc-bxivhb hfKSal"
      >
         Our Most Popular Categories
      </span>
    </h1>
    <section
      className="sc-ifAKCX iUhJXj"
    />
  </section>
</section>
`;
